src/util/psbt/serialize.rs

```diff
@@ -20,3 +26,62 @@ pub trait Deserialize: Sized {
 }
 
 impl_psbt_de_serialize!(Transaction);
+
+impl Serialize for Script {
+    fn serialize(&self) -> Vec<u8> {
+        self.to_bytes()
+    }
+}
+
+impl Deserialize for Script {
+    fn deserialize(bytes: &Vec<u8>) -> Result<Self, encode::Error> {
+        Ok(Self::from(bytes.clone()))
+    }
+}
+
+impl Serialize for PublicKey {
+    fn serialize(&self) -> Vec<u8> {
+        self.serialize().to_vec()
+    }
+}
+
+impl Deserialize for PublicKey {
+    fn deserialize(bytes: &Vec<u8>) -> Result<Self, encode::Error> {
+        let ref mut d = Cursor::new(&bytes[..]);
+
+        let serialized: [u8; 33] = Decodable::consensus_decode(d)?;
+        PublicKey::from_slice(&Secp256k1::new(), &serialized)
+            .map_err(|_| encode::Error::ParseFailed("invalid public key"))
+    }
+}
+
+impl Serialize for (Fingerprint, Vec<ChildNumber>) {
+    fn serialize(&self) -> Vec<u8> {
+        let mut rv: Vec<u8> = Vec::with_capacity(4 + 4 * (&self.1).len());
+
+        rv.append(&mut self.0.to_bytes().to_vec());
+
+        for cnum in self.1.iter() {
+            rv.append(&mut serialize(&u32::from(cnum.clone())))
+        }
+
+        rv
+    }
+}
+
+impl Deserialize for (Fingerprint, Vec<ChildNumber>) {
+    fn deserialize(bytes: &Vec<u8>) -> Result<Self, encode::Error> {
+        let ref mut d = Cursor::new(&bytes[..]);
+
+        let serialized: [u8; 4] = Decodable::consensus_decode(d)?;
+        let fprint: Fingerprint = Fingerprint::from(&serialized[..]);
+
+        let mut dpath: Vec<ChildNumber> = Default::default();
+
+        while let Ok(index) = Decodable::consensus_decode(d) {
```

Comment:
Should end when the cursor hits the end of the input, not just on parse error. (And a parse error should fail the decode.)